src/main/java/org/wildfly/security/credential/store/impl/KeyStoreCredentialStore.java

```diff
@@ -605,7 +605,7 @@ public void store(final String credentialAlias, final Credential credential, fin
                     case OneTimePassword.ALGORITHM_OTP_SHA1: {
                         decoder.startSequence();
                         final byte[] hash = decoder.decodeOctetString();
-                        final byte[] seed = decoder.decodeOctetString();
+                        final String seed = decoder.decodeOctetStringAsString(StandardCharsets.US_ASCII.name());
```

Comment:
Here's the other spot where we should switch to using encodeIA5String and decodeIA5String. Everything else looks good.